File: 1.js

let pow = Math.pow

function commit(v, r){
  p = 1133261442823444443917138265631

  return (BigInt(3) ** BigInt(v) % BigInt(p)) * (BigInt(5) ** BigInt(r) % BigInt(p)) % BigInt(p)
}

const r = Math.floor(Math.random() * 1000)


console.log(commit(5, r));

File: 1.py


from Crypto.Util.number import *



p = 1133261442823444443917138265631

def commit(v, r):
    return pow(3,v,p) * pow(5,r,p) % p

    #     v1, r1         v2, r2
a = commit(2, 6) * commit(5, 9)
b = commit(7, 15)
assert(a == b)

print(a, b)